### PR TITLE
feat: add video and animation to generating keys page

### LINKED ISSUE #16074: Onboarding - Generate Keys - implement designs
Current version of the Generating Keys is just a placeholder - 

implement as in the designs.

Should use a similar approach as the other screens for onboarding -
common background
content only transition animations

https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=2379-272834&t=YWqXodGUAGD3g8B5-4

### PR DESCRIPTION
**Pull request by J-Son89**
fixes: https://github.com/status-im/status-mobile/issues/16074

This pr adds the video for the generating keys page and a basic version of the animation that will be used. There is some future adjustments that are not defined yet from the design team but it will involve the animation moving off the page at the end. 
 
This will only work on whitelisted devices (currently iphones 11+) however for lower spec devices there is a fallback of 3 images. Further input from design team will eventually be needed for the lower end devices.

### CODE REVIEW

File: src/status_im2/contexts/onboarding/events.cljs [hunk 1]
```diff
@@ -111,12 +111,14 @@
...
                          :currentNetwork              config/default-network
                          :previewPrivacy              config/blank-preview?}]
-    {effect    request
-     :dispatch [:navigate-to :generating-keys]
-     :db       (-> db
-                   (dissoc :multiaccounts/login)
-                   (dissoc :auth-method)
-                   (assoc :onboarding-2/new-account? true))}))
+    {effect          request
+     :dispatch       [:navigate-to :generating-keys]
```
Comment (J-Son89) [inline src/status_im2/contexts/onboarding/events.cljs]: @briansztamfater is this `:dispatch` needed? 🤔

File: src/status_im2/contexts/onboarding/events.cljs [hunk 2]
```diff
@@ -111,12 +111,14 @@
...
+     :dispatch-later [{:ms       constants/onboarding-generating-keys-animation-duration-ms
```
Comment (J-Son89) [inline src/status_im2/contexts/onboarding/events.cljs]: waits until generate keys animation has finished, and then has a fallback check to ensure keys are definitely generated in which case the transition will be slightly delayed 👍

**Review by smohamedjavid — APPROVED** [review]
LGTM 🚀

File: src/status_im2/contexts/onboarding/events.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im2/contexts/onboarding/events.cljs]: I think it is, we immediately navigate to `:generating-keys` screen and then dispatch `:onboarding-2/navigate-to-identifiers` with delay.

File: src/status_im2/contexts/onboarding/generating_keys/style.cljs [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
-(def page-container
-  {:position         :absolute
-   :top              0
-   :bottom           0
-   :left             0
-   :right            0
+(defn page-container
+  [insets]
+  {:flex             1
+   :justify-content  :space-between
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/style.cljs:7]: Super minor but there is an extra space after `:justify-content`

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 4]
```diff
@@ -1,27 +1,152 @@
...
+   {:title (i18n/label :t/saving-keys-to-device)}])
+
+(defn keys-saved-title
+  []
+  [quo/title
+   {:title (i18n/label :t/keys-saved)}])
+
+(def first-transition-delay 2000)
+
+(def duration 500)
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: maybe `transition-duration-ms`

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 5]
```diff
@@ -1,27 +1,152 @@
...
+  []
+  [quo/title
+   {:title (i18n/label :t/saving-keys-to-device)}])
+
+(defn keys-saved-title
+  []
+  [quo/title
+   {:title (i18n/label :t/keys-saved)}])
+
+(def first-transition-delay 2000)
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: maybe `first-transition-delay-ms`

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 6]
```diff
@@ -1,27 +1,152 @@
...
+                                                        (js-obj "duration" duration
+                                                                "easing"   (:linear
+                                                                            reanimated/easings))))))
+
+(defn f-title
+  [insets generate-keys-opacity saving-keys-opacity keys-saved-opacity]
+  [rn/view
+   {:position :absolute
+    :top      (+ (if rn/small-screen? 62 112) (:insets insets))}
+
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: Check extra lines in this fn

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 7]
```diff
@@ -1,27 +1,152 @@
...
+                                (reanimated/with-timing 1
+                                                        (js-obj "duration" duration
+                                                                "easing"   (:linear
+                                                                            reanimated/easings))))))
+
+(defn f-title
+  [insets generate-keys-opacity saving-keys-opacity keys-saved-opacity]
+  [rn/view
+   {:position :absolute
+    :top      (+ (if rn/small-screen? 62 112) (:insets insets))}
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: maybe use a let binding for this value to improve readability

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 8]
```diff
@@ -1,27 +1,152 @@
...
+
+   [reanimated/view
+    {:style (reanimated/apply-animations-to-style
+             {:opacity generate-keys-opacity}
+             {:position :absolute})}
+    [generate-keys-title]]
+
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: extra line

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs [hunk 9]
```diff
@@ -1,27 +1,152 @@
...
+                                                                "easing"   (:linear
+                                                                            reanimated/easings)))))
+  (reanimated/set-shared-value
+   saving-keys-opacity
+   (reanimated/with-sequence
+    (reanimated/with-delay 2000
+                           (reanimated/with-timing 1
+                                                   (js-obj "duration" duration
+                                                           "easing"   (:linear reanimated/easings))))
+
```
Comment (briansztamfater) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: extra line

**Review by briansztamfater — APPROVED** [review]
Added some minor comments to improve code readability, but overall looks good to me!

**Comment by J-Son89** [comment]
> Added some minor comments to improve code readability, but overall looks good to me!

Thanks @briansztamfater, will adjust!

**Comment by qoqobolo** [comment]
Hey @J-Son89, could you resolve the conflict and rebase, please?

**Comment by J-Son89** [comment]
@qoqobolo - this pr is ready to test again :)

**Comment by qoqobolo** [comment]
Thanks @J-Son89!
Looks cool :) 

One thing: it seems to be placed a little high compared to the design, too close to the header. Do you think we can move it a little lower? This is especially noticeable on devices with a larger screen (i.e. it is not so noticeable on iPhone SE ).

PR

Screenshot 2023-07-10 at 16 10 05

Design 

Screenshot 2023-07-10 at 16 10 42

**Comment by qoqobolo** [comment]
@J-Son89 and look at tests please, something's wrong with them again :)

**Comment by J-Son89** [comment]
@qoqobolo tests are passing now - it fixed with a rebase :) 
also I made the animation slightly smaller in scale as that's how the designs show it and then pushed them down lower on the screen as you suggested :)

**Comment by qoqobolo** [comment]
Thanks @J-Son89, LGTM!
PR can be merged.

**Pull request merged**